src/libraries/Common/src/Interop/Interop.ICU.cs

```diff
@@ -11,6 +12,9 @@ internal static partial class Globalization
         [DllImport(Libraries.GlobalizationNative, EntryPoint = "GlobalizationNative_LoadICU")]
         internal static extern int LoadICU();
 
+        [DllImport(Libraries.GlobalizationNative, EntryPoint = "GlobalizationNative_InitICUFunctions")]
+        internal static extern void InitICU(IntPtr icuuc, IntPtr icuin, string version, string? suffix);
```

Comment:
Both calls to this have spans and are ToString'ing to get strings, which the runtime will then need to marshal as char*.  Maybe instead:
```C#
[DllImport(Libraries.GlobalizationNative, EntryPoint = "GlobalizationNative_InitICUFunctions")]
private static extern unsafe void InitICU(IntPtr icuuc, IntPtr icuin, char* version, char* suffix);

internal static unsafe void InitICU(IntPtr icuuc, IntPtr icuin, ReadOnlySpan<char> version, ReadOnlySpan<char> suffix)
{
    fixed (char* versionPtr = version, suffixPtr = suffix)
        InitIcu(icuuc, icuin, versionPtr, suffixPtr);
}
```
and then remove the ToString()s from the call sites?

Reply:
Discussion offline with @safern highlighted that the char*s need to be null terminated, so this won't work.

Reply:
as we chatted offline, I need the version to be null terminated since I parse the version natively to find the exported symbols using `sscanf` so I can't just pass down the ptr from the Span, using string works as it is null terminated.